### PR TITLE
Add base GitHub Action for the Pull Request Merger

### PR DESCRIPTION
**Pull request by agorararmard**
In collaboration with @ax3ghazy, we created this workflow for the Pull Request Merger. This PR is for opening the discussion on the sample implementation of the workflow.

Current Workflow when a PR is submitted:
- It has to be reviewed by one of the repo collaborators.
- Get labeled ready-to-merge.
- The Action will get the PR as a patch.
- The Action will apply the patch to all version branches merging upward whenever applying is possible.
- The Action will reset the master to the latest version branch.
- The Action will close the PR and comment to the contributor that the merger added it to the repo.

### CODE REVIEW

**Review by mithro — COMMENTED** [review]
Can we move this to a separate repository and publish it as GitHub Action?

**Comment by mithro** [comment]
(Or maybe it should be an action in the https://github.com/google/skywater-pdk repository?)

**Comment by agorararmard** [comment]
@mithro: I think the behavior is too specific to be a useful action for other repos, am I right? But we can have this as a separate repo and submodule it in all libraries if that would be more convenient. However, I think we better discuss this in our next meeting in some more detail.

**Comment by mithro** [comment]
Well more useful for all the skywater-pdk modules?

**Comment by agorararmard** [comment]
@mithro: Why not. 
I created an action here https://github.com/marketplace/actions/pull-request-merger-on-releases and updated this PR to use it.

**Comment by mithro** [comment]
Where is the source now?

File: .github/workflows/pull_request_merger.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: master
+          fetch-depth: '50'
+
+      - name: Skywater PDK Libraries Pull Request Merger on Release Branches
+        uses: agorararmard/skywater-pdk-modules-pull-request-backporter-action@v2
```
Comment (mithro) [inline .github/workflows/pull_request_merger.yml:18]:
```suggestion
        uses: google/skywater-pdk-modules-actions/modules-pr-backporter@main
```

Comment (agorararmard) [reply .github/workflows/pull_request_merger.yml:18]: as per our discussion two weeks ago, this PR is obsolete. We will be getting the history for this repo from: https://github.com/efabless/skywater-pdk-libs-sky130_fd_sc_hd

**Pull request closed without merging**